src/sql/src/plan/typeconv.rs

```diff
@@ -174,10 +174,24 @@ lazy_static! {
             (Oid, RegType) => Assignment: CastOidToRegType,
 
             // REGPROC
-            (RegProc,Oid) => Implicit: CastRegProcToOid,
+            (RegProc, Oid) => Implicit: CastRegProcToOid,
+            (RegProc, String) => Explicit: sql_impl_cast("(
+                SELECT DISTINCT(COALESCE(t.name, v.x::pg_catalog.text))
```

Comment:
The distinct here is wrong, it's supposed to error. See postgres:

```
mjibson=#  select 'generate_series'::regproc;
ERROR:  more than one function named "generate_series"
LINE 1: select 'generate_series'::regproc;
```

`generate_series` returning multiple entries is the actual bug, because it incorrectly re-uses the same oid.

Reply:
same behavior in materialize.
```
materialize=> select 'generate_series'::regproc;
ERROR:  Evaluation error: more than one record produced in subquery
```

`distinct` is required to support this query `select oid, oid::regproc::text from (select oid from mz_catalog.mz_functions);`. which is an equivalent of PG `select oid, oid::regproc::text from (select oid from pg_catalog.pg_proc) as foo;`

Reply:
it seems that PG does not have duplicates.
```
martin=# select cnt, oid from (select count(*) as cnt, oid from pg_catalog.pg_proc group by oid) as foo where cnt>1;
 cnt | oid 
-----+-----
(0 rows)
```

and MZ does:
```
materialize=> select cnt, oid from (select count(*) as cnt, oid from mz_catalog.mz_functions group by oid) where cnt>1;
 cnt | oid  
-----+------
   2 | 1066
(1 row)
```

Reply:
confirming that one of the types is incorrect.
```
materialize=> select * from mz_catalog.mz_types where oid in (20,23);
  id   | oid | schema_id | name 
-------+-----+-----------+------
 s1002 |  20 |         2 | int8
 s1003 |  23 |         2 | int4
(2 rows)

materialize=> select * from mz_catalog.mz_functions where name like '%generate_series%';
  id   | oid  | schema_id |      name       |       arg_ids       | variadic_id 
-------+------+-----------+-----------------+---------------------+-------------
 s2040 | 1067 |         2 | generate_series | {s1003,s1003}       | 
 s2040 | 1069 |         2 | generate_series | {s1002,s1002}       | 
 s2040 | 1066 |         2 | generate_series | {s1002,s1002,s1002} | 
 s2040 | 1066 |         2 | generate_series | {s1003,s1003,s1003} | 
(4 rows)
```

```
martin=# select oid,typname from pg_catalog.pg_type where oid in (20, 23);
 oid | typname 
-----+---------
  20 | int8
  23 | int4
(2 rows)
martin=# select oid, proname, proargtypes from pg_catalog.pg_proc where proname like '%generate_series%' and pronargs=3;
 oid  |     proname     |  proargtypes   
------+-----------------+----------------
 1066 | generate_series | 23 23 23
 1068 | generate_series | 20 20 20
 3259 | generate_series | 1700 1700 1700
  938 | generate_series | 1114 1114 1186
  939 | generate_series | 1184 1184 1186
(5 rows)
```

I will fix this in a separate PR.

Reply:
fixed here. https://github.com/MaterializeInc/materialize/pull/9199. Once I rebase, I will remove the distinct